### PR TITLE
Sort diag_states_summary.php by states

### PR DESCRIPTION
**Pull request by teunis90**
- [-] Redmine Issue: No redmine issue
- [x] Ready for review

In my human opinion the diag_states_summary.php page is more helpful when the entries are sorted by states ($seen). In this case you see the top talkers on top of the page.

Please merge if helpful.

### CODE REVIEW

File: src/usr/local/www/diag_states_summary.php [hunk 1]
```diff
@@ -129,8 +129,11 @@ function build_port_info($portarr, $proto) {
 }
 
 function print_summary_table($label, $iparr, $sort = TRUE) {
+    foreach ($iparr as $ip => $ipinfo) {
+        $iparr[$ip]['ip'] = $ip;
```
Comment (jim-p) [inline src/usr/local/www/diag_states_summary.php]: Why do this? The IP address is already there as a key. This will slow the page down and consume a lot more memory with large state tables.

File: src/usr/local/www/diag_states_summary.php [hunk 2]
```diff
@@ -157,7 +160,7 @@ function print_summary_table($label, $iparr, $sort = TRUE) {
 						</tr>
 					</thead>
 					<tbody>
-<?php foreach ($iparr as $ip => $ipinfo):
+<?php foreach ($iparr as $index => $ipinfo):
```
Comment (jim-p) [inline src/usr/local/www/diag_states_summary.php]: See above, this appears to be an unnecessary change

File: src/usr/local/www/diag_states_summary.php [hunk 3]
```diff
@@ -167,9 +170,10 @@ function print_summary_table($label, $iparr, $sort = TRUE) {
 	}
 ?>
 						<tr>
-							<td<?= $rowSpan ?>><?=$ip;?></td>
+							<td<?= $rowSpan ?>><?=$ipinfo['ip'];?></td>
```
Comment (jim-p) [inline src/usr/local/www/diag_states_summary.php]: See above, this appears to be an unnecessary change

File: src/usr/local/www/diag_states_summary.php [hunk 4]
```diff
@@ -129,8 +129,11 @@ function build_port_info($portarr, $proto) {
 }
 
 function print_summary_table($label, $iparr, $sort = TRUE) {
+    foreach ($iparr as $ip => $ipinfo) {
+        $iparr[$ip]['ip'] = $ip;
+    }
 	if ($sort) {
-		uksort($iparr, "sort_by_ip");
+		usort($iparr, "sort_by_seen");
```
Comment (jim-p) [inline src/usr/local/www/diag_states_summary.php]: This should be @uasort()@, not @usort()@.

**Comment by teunis90** [comment]
@jim-p I filed an issue as requested: Redmine: 9152
I was not aware of `uasort()`, that's definitely an improvement of this little code change.

Please merge.

**Comment by sbeaver-netgate** [comment]
While useful, I don't think we should impose a particular order without user input. Could you provide a checkbox or select control to allow the user to choose which order the states are listed in?

**Comment by teunis90** [comment]
I agree, but sorry, that is not something where I would have time for in the near future.

Since the current order doesn't make any sense at all, I do think a static order is 'better' than the current implementation.

**Comment by rbgarga** [comment]
I've opened a redmine ticket to make columns sortable

https://redmine.pfsense.org/issues/9718
